tokenserver/verifiers.py

```diff
@@ -144,3 +159,71 @@ def verify(self, assertion, audience=None):
             if issuer not in self.allowed_issuers:
                 raise InvalidIssuerError("Issuer not allowed: %s" % (issuer,))
         return data
+
+
+class RemoteOAuthVerifier(object):
+    """A Verifier for FxA OAuth tokens that posts to a verifiaction service.
+
+    This verifier uses the remote FxA OAuth verification service to accept
+    OAuth tokens, and translates the returned data into something that
+    approximates the information provided by BrowserID.
+
+    In order to meet tokenserver's expectation that users are identified
+    with an email address, it combines the FxA uid with the hostname of
+    the corresponding FxA BrowserID issuer.  For non-standard FxA hosting
+    setups this might require it to dynamically discover the BrowserID
+    issuer by querying the OAuth verifier's configuration.
+    """
+    implements(IOAuthVerifier)
+
+    def __init__(self, server_url=None, issuer=None, timeout=None,
+                 scope='https://identity.mozilla.com/apps/oldsync'):
+        if timeout is None:
```

Comment:
Do you have instances where you pass `None` as the value, or is this more to prevent library accidents?

Reply:
I'll make it a plain old default arg.